### PR TITLE
Use OS temporary folder as the download cache.

### PR DESCRIPTION
**Pull request by vjrantal**
A part of the installation is the download of the required version of the
JXCore Cordova plugin (currently ~90MB download) and unpacking that with jx
before adding it as Cordova plugin using Cordova commands.

This commit implements caching of the unpacked plugin folder so that the
overall installation process becomes faster. The OS temporary folder is used
for caching instead of any Thali-app folder so that full benefits of the cache
are received even if the app-folder is re-generated (this is the case with CI
scripts, for example).

This commit also adds some catches to the promise-chains so that in case of
errors, it is easier to see where the issue was.

Relates to #192.

[Review on Reviewable](https://reviewable.io/reviews/thaliproject/thali_cordovaplugin/251)

### CODE REVIEW

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (e5434ba)

**Comment by yaronyg** [comment]
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

<sup>**[thali/install/install.js, line 218 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1eF1P4qcywW6Swh933)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/e5434ba0d24cff1cba5939472ebde8eedbc082bc/thali/install/install.js#L218)):</sup>
Couldn't we just get rid of the basedir/jxcore and just run everything out of the cache folder? Or is the idea that we will augment the code so when we are running on CI the cache directory will be remote and therefore we need to copy it locally and hence the basedir? It's not a big deal but I want to make sure I actually understand what the code is doing.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251)

**Comment by ThaliTester** [comment]
Test (Success) 48899917 build is completed (e5434ba)

See https://github.com/ThaliTester/TestResults/tree/48899917e5434ba_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by vjrantal** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

<sup>**[thali/install/install.js, line 218 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1eF1P4qcywW6Swh933)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/e5434ba0d24cff1cba5939472ebde8eedbc082bc/thali/install/install.js#L218)):</sup>
The script had logic to store the version of the installed plugin onto a specific file that is stored in the "app context" rather than the "cache context". That seems to be the only reason why the baseDir/jxcore folder is still required there.

We could improve this so that we are not writing the plugin version to a file, but rather using Cordova commands to check what is the actual version of the installed plugin from Cordova perspective. Then, this logic would also work if developer has manually managed the plugin with Cordova commands.

I think this PR could go in as is and we could add above improvement separately later.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251)

**Comment by yaronyg** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

<sup>**[thali/install/install.js, line 218 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1eF1P4qcywW6Swh933)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/e5434ba0d24cff1cba5939472ebde8eedbc082bc/thali/install/install.js#L218)):</sup>
Or we could get rid of baseDir all together and just check the cache directory to see if the version we want is there and then use it. If not then we can download it. This script is really only intended to be used 'clean'. It isn't intended to work in terms of updating an existing ThaliTest folder. So I'm not too worried about talking to cordova.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251)

**Comment by vjrantal** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

<sup>**[thali/install/install.js, line 218 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1eF1P4qcywW6Swh933)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/e5434ba0d24cff1cba5939472ebde8eedbc082bc/thali/install/install.js#L218)):</sup>
I can remove the version-checking functionality from the script and then we can also get rid of that directory.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251)

**Comment by ThaliTester** [comment]
Test 48899917e5434ba(e5434ba) has failed

See https://github.com/ThaliTester/TestResults/tree/48899917e5434ba_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the fail logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (84e2e77)

**Comment by ThaliTester** [comment]
Test (Fail) 48899917 build is completed (84e2e77)

See https://github.com/ThaliTester/TestResults/tree/4889991784e2e77_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (99af7f0)

**Comment by ThaliTester** [comment]
Test (Fail) 48899917 build is completed (99af7f0)

See https://github.com/ThaliTester/TestResults/tree/4889991799af7f0_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (eeeb394)

**Comment by ThaliTester** [comment]
Test (Fail) 48899917 build is completed (eeeb394)

See https://github.com/ThaliTester/TestResults/tree/48899917eeeb394_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (c5c0947)

**Comment by ThaliTester** [comment]
Test (Fail) 48899917 build is completed (c5c0947)

See https://github.com/ThaliTester/TestResults/tree/48899917c5c0947_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (dd7ac55)

**Comment by ThaliTester** [comment]
Test (Fail) 48899917 build is completed (dd7ac55)

See https://github.com/ThaliTester/TestResults/tree/48899917dd7ac55_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (f1a621f)

**Comment by ThaliTester** [comment]
Test (Fail) 48899917 build is completed (f1a621f)

See https://github.com/ThaliTester/TestResults/tree/48899917f1a621f_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (7b5fcb3)

**Comment by ThaliTester** [comment]
Test (Fail) 48899917 build is completed (7b5fcb3)

See https://github.com/ThaliTester/TestResults/tree/488999177b5fcb3_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (4bf7175)

**Comment by ThaliTester** [comment]
Test (Fail) 48899917 build is completed (4bf7175)

See https://github.com/ThaliTester/TestResults/tree/488999174bf7175_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (d0ecf82)

**Comment by ThaliTester** [comment]
Test (Success) 48899917 build is completed (d0ecf82)

See https://github.com/ThaliTester/TestResults/tree/48899917d0ecf82_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (0dfe2c5)

**Comment by ThaliTester** [comment]
Test (Success) 48899917 build is completed (0dfe2c5)

See https://github.com/ThaliTester/TestResults/tree/488999170dfe2c5_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by ThaliTester** [comment]
Test 48899917d0ecf82(d0ecf82) has failed

See https://github.com/ThaliTester/TestResults/tree/48899917d0ecf82_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the fail logs

**Comment by ThaliTester** [comment]
Test 488999170dfe2c5(0dfe2c5) has failed

See https://github.com/ThaliTester/TestResults/tree/488999170dfe2c5_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the fail logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 2. task. (f231b24)

**Comment by ThaliTester** [comment]
Test (Success) 48899917 build is completed (f231b24)

See https://github.com/ThaliTester/TestResults/tree/48899917f231b24_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by vjrantal** [comment]
Review status: 0 of 2 files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---

<sup>**[thali/install/install.js, line 218 \[r1\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1eF1P4qcywW6Swh933)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/e5434ba0d24cff1cba5939472ebde8eedbc082bc/thali/install/install.js#L218)):</sup>
Done.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251)

**Comment by vjrantal** [comment]
For the record, I today verified that the installation works also when installing Thali on a Windows machine.

Also, the experiments done today in the CI indicate the scp-based "remote cache" also works as expected in the CI environment.

**Comment by yaronyg** [comment]
10/28 - This ended up eating the whole day because of issues with SCP running on CI. There was a problem with CI that Oguz eventually had to fix manually.

**Comment by yaronyg** [comment]
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

<sup>**[thali/install/install.js, line 1 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1jC-4KiLuxEB5zF6W7)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/f231b24ccbf77c974df69078cec481d3eb7929d0/thali/install/install.js#L1)):</sup>
Please do me a favor and go [here](https://github.com/thaliproject/Thali_CordovaPlugin/blob/fix-192/thali/install/install.js) and stare at the file for a second. It's 419 lines long of detailed install work.

Please look at [this function](https://github.com/thaliproject/Thali_CordovaPlugin/blob/fix-192/thali/install/install.js#L208) in particular. It's 155 lines long. The general rule of thumb is that the human brain loses context after 60 lines or so (aka a single page of content).

This file, at a minimum, really needs to be refactored. Especially the `fetchAndInstallJxCoreCordovaPlugin`function. It needs to be broken into discrete parts with sensible names so someone can read it all the way down with a sense of what it is doing and then jump to the sub-functions to see the details.

We are going to be messing with this code for a very long time so I believe it's worthwhile to make sure the code is nicely factored and maintainable.


---

<sup>**[thali/install/install.js, line 241 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1jAB_wkTiCbmaunBrj)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/f231b24ccbf77c974df69078cec481d3eb7929d0/thali/install/install.js#L241)):</sup>
NIT - It is generally considered good style in a situation like this to put a return after the resolve. Then get rid of the else clause and promote the content in the else clause to the left. The idea is that it makes it unambiguous that the function must exit at the end of the if. Otherwise someone can be forgiven for putting something after the if/else clause.


---

<sup>**[thali/install/install.js, line 244 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1jApR_ltMl6PH1PBZ8)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/f231b24ccbf77c974df69078cec481d3eb7929d0/thali/install/install.js#L244)):</sup>
NIT - Personally I find having all the SCP related variables and functionality mixed in with everything else makes the code hard to follow. I would probably have just created a SCP function that takes jxCoreVersionNumber as an argument and returns a promise.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251)

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 5. task. (d9ae287)

**Comment by ThaliTester** [comment]
Test (Success) 48899917 build is completed (d9ae287)

See https://github.com/ThaliTester/TestResults/tree/48899917d9ae287_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by vjrantal** [comment]
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

<sup>**[thali/install/install.js, line 1 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1jC-4KiLuxEB5zF6W7)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/f231b24ccbf77c974df69078cec481d3eb7929d0/thali/install/install.js#L1)):</sup>
I 100% agree - this file is a monster! :)

When I started doing this work, it took me quite some time to figure out all the functionality implemented in this file. Also, found out several code paths that didn't work at all!

Like you say, the file needs to be refactored and broken down and in addition, there needs to be tests for the discrete parts.

The reason I didn't invest time into above *before* submitting this PR was the following:

1. I wanted to get the build reliable (and faster) as soon as possible, because the issues with it currently eats time from all developers.
2. I was not sure if this is a long-term solution for the installation process overall.

To elaborate on point 2. above, what I was thinking is that currently, the installation process of the Cordova parts is driven by the thali npm package even though from software architecture perspective, the dependency is the other way around (i.e., the Cordova parts depend on the thali npm package). The implication is that, for example, you can't install do `npm install thali` on an empty folder, because the npm package assumes that it finds the Cordova parts from the installation folder.

In the future, a better approach might be that the Thali Cordova plugin installs the thali npm package. For a Cordova app developer, the installation of the Thali Cordova plugin would end up looking like installation of any other Cordova plugin, i.e., simply something like `cordova plugin add org.thaliproject.p2p`. That command would then take care of the installation of all dependencies the Thali Cordova plugin has (including the thali npm package and the jxcore Cordova plugin). This would also allow developing the thali npm package into a direction that it can be tested and leveraged also from a non-Cordova app, for example, a headless app running on a Rasperry Pi that wants to "talk thali".


---

<sup>**[thali/install/install.js, line 241 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1jAB_wkTiCbmaunBrj)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/f231b24ccbf77c974df69078cec481d3eb7929d0/thali/install/install.js#L241)):</sup>
Done.


---

<sup>**[thali/install/install.js, line 244 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1jApR_ltMl6PH1PBZ8)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/f231b24ccbf77c974df69078cec481d3eb7929d0/thali/install/install.js#L244)):</sup>
Done.

I also mentioned in the commit message that the individual pieces could be moved onto their own modules (own files), but I didn't yet start creating new files. This could be done in a larger refactoring in the future.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251)

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 6. task. (5eacbd2)

**Comment by ThaliTester** [comment]
Test (Success) 48899917 build is completed (5eacbd2)

See https://github.com/ThaliTester/TestResults/tree/488999175eacbd2_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by ThaliTester** [comment]
Test 48899917f231b24(f231b24) has failed

See https://github.com/ThaliTester/TestResults/tree/48899917f231b24_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the fail logs

**Comment by ThaliTester** [comment]
Test 48899917d9ae287(d9ae287) has failed

See https://github.com/ThaliTester/TestResults/tree/48899917d9ae287_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the fail logs

**Comment by ThaliTester** [comment]
Test 488999175eacbd2(5eacbd2) has failed

See https://github.com/ThaliTester/TestResults/tree/488999175eacbd2_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the fail logs

**Comment by vjrantal** [comment]
Review status: 1 of 2 files reviewed at latest revision, 3 unresolved discussions.

---

<sup>**[thali/install/install.js, line 1 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1jC-4KiLuxEB5zF6W7)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/f231b24ccbf77c974df69078cec481d3eb7929d0/thali/install/install.js#L1)):</sup>
I want to correct myself on the previous comment related to "from software architecture perspective, the dependency is the other way around", because currently, that is not true. I guess what I tried to say was that there might be certain benefits if in the long run, the direction would be towards having the dependency that way since then, the development experience would become more familiar for Cordova app developers and the thali npm package could be leveraged more without Cordova.

Overall, I'd tend to agree that it is worthwhile to do some refactoring and hardening for the current installation flow, but in my opinion, that work should not block this PR, but should be created as it's own work item.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251)

**Comment by yaronyg** [comment]
Reviewed 1 of 1 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

<sup>**[thali/install/install.js, line 1 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1jC-4KiLuxEB5zF6W7)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/f231b24ccbf77c974df69078cec481d3eb7929d0/thali/install/install.js#L1)):</sup>
Fair enough. I opened #265 to cover this.

As for having the install the wrong way around, I don't believe it's that simple.

If you think of Thali as being a Node.js application (which it is) then you need to think about this from the perspective of a Node.js developer. That developer is writing a bunch of code of which Thali is only a part. They will have their own app.js with a bunch of libraries, one of them is Thali. As a node.js developer how would you expect to install Thali? Via NPM of course!

So the question is - who do we optimize for? A node.js developer who is hosting their code in Cordova or a Cordova developer who is using Thali?

Depending on who your primary customer is that will determine if it makes sense to install Thali as a Cordova plugin or as a NPM package.

Also keep in mind that the Thali code runs just fine outside of Cordova. So today, for example, I can install Thali on my desktop to an App and it will install just fine. The Mobile calls will fail but that is to be expected. But once we have your SSDP code that Thali on the desktop app will be fully functional. So I don't think we want to lose the ability to install Thali via NPM or we lose that desktop capability.

In an ideal world we would just support both ways. And in the long run we probably will. I strongly suspect that Thali will have a non-trivial contingent of programmers who will only use it as a Cordova plugin that they control from the webview via the REST interface.  But I also suspect we will have people who are using Thali outside of Cordova and also who use it inside of Cordova but as apart of their own node.js app that is doing other interesting things. So there will be an audience for both approaches.

But I had to start somewhere so I decided to start with NPM. It seems to give us the most bang for the buck. 


---

<sup>**[thali/install/install.js, line 206 \[r4\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1p_wBBapxBXMn7ioTT)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/5eacbd262143f228e7345616d58c69275b46fe85/thali/install/install.js#L206)):</sup>
Please don't do this. Self calling functions are super painful to read and follow. They cause weird indenting. It's just not happy. Please define the functions as functions and just call them normally. This kind of nested structure causes migraines. If you really want it to be completely stand alone than please move it to its own file and then require it here and use module.exports in the new file.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251)

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 1. task. (5eacbd2)

**Comment by yaronyg** [comment]
Waiting for Ville's response (and I thought we fixed CI So it wouldn't launch a new test round just because we changed the issue owner?!?!?)

**Comment by obastemur** [comment]
See.. I need to move the in memory DB to JS file DB. It has just triggered the test.

**Comment by ThaliTester** [comment]
Test (Success) 48899917 build is completed (5eacbd2)

See https://github.com/ThaliTester/TestResults/tree/488999175eacbd2_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by ThaliTester** [comment]
Test 488999175eacbd2(5eacbd2) has failed

See https://github.com/ThaliTester/TestResults/tree/488999175eacbd2_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the fail logs

**Comment by ThaliTester** [comment]
PR is added to the queue for testing as 3. task. (7028376)

**Comment by ThaliTester** [comment]
Test (Success) 48899917 build is completed (7028376)

See https://github.com/ThaliTester/TestResults/tree/488999177028376_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the logs

**Comment by vjrantal** [comment]
Review status: 1 of 3 files reviewed at latest revision, 2 unresolved discussions.

---

<sup>**[thali/install/install.js, line 1 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1jC-4KiLuxEB5zF6W7)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/f231b24ccbf77c974df69078cec481d3eb7929d0/thali/install/install.js#L1)):</sup>
I agree with everything said above :)

The reason I raised this discussion was that I think these might be questions that people approaching Thali "from outside" might think about.

Currently, Thali's developer offering might be seen by someone to target Cordova developers, because the "entry point" is a Cordova plugin. However, as you say above, the current API is for Node.js developers.

When I first started reading about Thali and the API documentation from https://github.com/thaliproject/Thali_CordovaPlugin#documentation I got the impression that those are APIs for Cordova developers, but they aren't. The JavaScript APIs mentioned in the linked documentation aren't available in the "Cordova context", but in the "JXCore context".

I agree that the current installation flow makes sense and serves better Node.js developers and should be polished. So, totally agree on https://github.com/thaliproject/Thali_CordovaPlugin/issues/265.

By the way, what I meant by not being to install thali npm package without Cordova is this:

```
$ mkdir -p /tmp/some/random/folder && cd /tmp/some/random/folder
$ npm install thali
\
> thali@2.0.1 postinstall /private/tmp/some/random/folder/node_modules/thali
> jx installCordovaPlugin.js

Downloaded io.jxcore.node.jx
Failed with - Error: Command failed: Current working directory is not a Cordova-based project.
```

However, above is easy to fix if this is seen as a "supported scenario".


---

<sup>**[thali/install/install.js, line 206 \[r4\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1p_wBBapxBXMn7ioTT)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/5eacbd262143f228e7345616d58c69275b46fe85/thali/install/install.js#L206)):</sup>
Agreed.

I made the "encapsulation" for preparation to move the scp-logic onto a separate file, but better to do that right away. I have pushed a commit that does this.


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251)

**Comment by ThaliTester** [comment]
Test 488999177028376(7028376) has failed

See https://github.com/ThaliTester/TestResults/tree/488999177028376_Use_OS_temporary_folder_as_the_download_cache__vjrantal/ for the fail logs

**Comment by yaronyg** [comment]
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

<sup>**[thali/install/install.js, line 1 \[r2\]](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251#-K1jC-4KiLuxEB5zF6W7)** ([raw file](https://github.com/thaliproject/thali_cordovaplugin/blob/f231b24ccbf77c974df69078cec481d3eb7929d0/thali/install/install.js#L1)):</sup>
Yup, the fix is called for in #270 


---


Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/thaliproject/thali_cordovaplugin/251)
